### PR TITLE
fix last_active_ago timestamps, tab completion ordering, and implement currently_active semantics

### PR DESCRIPTION
**Pull request by ara4n**

### CODE REVIEW

File: src/TabComplete.js [hunk 1]
```diff
@@ -24,10 +24,10 @@ const KEY_WINDOWS = 91;
 //
 // Capturing group containing the start
 // of line or a whitespace char
-//     \_______________       __________Capturing group of 1 or more non-whitespace chars
+//     \_______________       __________Capturing group of 0 or more non-whitespace chars
 //                    _|__  _|_         followed by the end of line
 //                   /    \/   \
-const MATCH_REGEX = /(^|\s)(\S+)$/;
+const MATCH_REGEX = /(^|\s)(\S*)$/;
```
Comment (richvdh) [inline src/TabComplete.js]: why is this changing? it scares me

File: src/components/views/rooms/EntityTile.js [hunk 2]
```diff
@@ -88,6 +88,7 @@ module.exports = React.createClass({
                     <img className="mx_EntityTile_chevron" src="img/member_chevron.png" width="8" height="12"/>
                     <div className="mx_EntityTile_name_hover">{ this.props.name }</div>
                     <PresenceLabel activeAgo={this.props.presenceActiveAgo}
+                        currentlyActive={this.props.presenceCurrentlyActive}
```
Comment (richvdh) [inline src/components/views/rooms/EntityTile.js]: `presenceCurrentlyActive` needs adding to propTypes with a comment descripbing wtf it does

File: src/TabComplete.js (hunk 1 above)
Comment (ara4n) [inline src/TabComplete.js]: as the commit says, it lets you hit tab on an empty textarea to start tab-completing the most recently active nick

File: src/components/views/rooms/EntityTile.js (hunk 2 above)
Comment (ara4n) [inline src/components/views/rooms/EntityTile.js]: done.

File: src/components/views/rooms/UserTile.js [hunk 3]
```diff
@@ -38,9 +38,8 @@ module.exports = React.createClass({
         var active = -1;
 
         // FIXME: make presence data update whenever User.presence changes...
-        active = (
-            (Date.now() - (user.lastPresenceTs - user.lastActiveAgo)) || -1
-        );
+        active = member.user.lastActiveAgo ?
```
Comment (dbkr) [inline src/components/views/rooms/UserTile.js]: Where does `member` come from here?

Comment (ara4n) [inline src/components/views/rooms/UserTile.js]: oooooooops, good catch; thanks

**Comment by ara4n** [comment]
@dbkr PTAL

**Comment by dbkr** [comment]
lgtm

**Pull request merged**